lido/validate_keys.py

```diff
@@ -18,6 +18,7 @@ def validate_key(data: t.Dict) -> t.Optional[bool]:
 
     key = data["key"]
     withdrawal_credentials = data["withdrawal_credentials"]
+    chain_id = data["chain_id"]
```

Comment:
We shouldn't take provided chain_id in input data - we need to use w3's chain. For example, key checker on Mainnet should never accept Goerli keys

Reply:
Yeah, you're right. I decided that passing the whole web3 object into the validate_key function just to get the chain_id is kinda redundant. For example, with the ability to pass chain_id instead of web3, we can use the validate_key function even without a network connection. If such a use case is not relevant, I'll revert the validate_key interface to the original one.
Also, it is possible that the user could pass Goerli keys (for example, from the database) while connecting to the Mainnet (e.g. because of configuration error). So I don't think that passing web3 instead of chain_id could prevent this kind of errors.

Reply:
Yep, that's a completely valid point - exactly why I wrote it the same way initially :)

I've tried implementing a strict/loose mode switch but it only complicated things without providing a use case to us.

Did not understand the second paragraph - Goerli keys should always fail validation if a non-Goerli network is connected. The real use case is that we have a key checker and key submitter with two variations (testnet and mainnet) and we should always deny and mark as invalid testnet keys on mainnet and vice versa.

Reply:
That's exactly what I'm talking about :)

The second paragraph is not about the proper behavior but about the mistake that could be made by the user. The sense is that if the function gets chain_id by itself, one cannot pass the wrong chain_id but still able to pass the wrong web3, inappropriate to the key one wants to validate (of course, in that scenario, the key would be marked as invalid). So the mistake of an inappropriate chain for key validation doesn't disappear. 

But you're right, passing web3 instead of chain_id reduces the probability of such a mistake, so I'll change the validate_key interface to the original one.

Reply:
Oh, got it, thank you!